##### Quels facteurs sont associés au comportement d'achat, et comment exploiter les résultats pour améliorer les ventes et la fidélisation ?

# 1. Préparation et nettoyage des données

In [6]:
import pandas as pd 
df=pd.read_csv("customer_shopping_behavior.csv", sep=",")



In [7]:
df.columns

Index(['Customer ID', 'Age', 'Gender', 'Item Purchased', 'Category',
       'Purchase Amount (USD)', 'Location', 'Size', 'Color', 'Season',
       'Review Rating', 'Subscription Status', 'Shipping Type',
       'Discount Applied', 'Promo Code Used', 'Previous Purchases',
       'Payment Method', 'Frequency of Purchases'],
      dtype='str')

In [9]:
df["Frequency of Purchases"].unique()

<StringArray>
[   'Fortnightly',         'Weekly',       'Annually',      'Quarterly',
      'Bi-Weekly',        'Monthly', 'Every 3 Months']
Length: 7, dtype: str

In [13]:
frequency_mapping = {
    "Weekly": 52,
    "Fortnightly": 26,
    "Bi-Weekly": 26,
    "Monthly": 12,
    "Quarterly": 4,
    "Every 3 Months": 4,
    "Annually": 1
}
df["Purchase_Frequency_Num"] = df["Frequency of Purchases"].map(frequency_mapping)

In [5]:
#Combien d’achats avons-nous dans chaque catégorie
df["Category"].value_counts()

Category
Clothing       1737
Accessories    1240
Footwear        599
Outerwear       324
Name: count, dtype: int64

In [11]:
df["Age_Group"] = pd.cut(
    df["Age"],
    bins=[17, 25, 35, 45, 55, 100],
    labels=["18-25", "26-35", "36-45", "46-55", "56+"]
)

In [16]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 3900 entries, 0 to 3899
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   Customer ID             3900 non-null   int64   
 1   Age                     3900 non-null   int64   
 2   Gender                  3900 non-null   str     
 3   Item Purchased          3900 non-null   str     
 4   Category                3900 non-null   str     
 5   Purchase Amount (USD)   3900 non-null   int64   
 6   Location                3900 non-null   str     
 7   Size                    3900 non-null   str     
 8   Color                   3900 non-null   str     
 9   Season                  3900 non-null   str     
 10  Review Rating           3863 non-null   float64 
 11  Subscription Status     3900 non-null   str     
 12  Shipping Type           3900 non-null   str     
 13  Discount Applied        3900 non-null   str     
 14  Promo Code Used         3900 non-nu

In [17]:
df.isnull().sum()

Customer ID                0
Age                        0
Gender                     0
Item Purchased             0
Category                   0
Purchase Amount (USD)      0
Location                   0
Size                       0
Color                      0
Season                     0
Review Rating             37
Subscription Status        0
Shipping Type              0
Discount Applied           0
Promo Code Used            0
Previous Purchases         0
Payment Method             0
Frequency of Purchases     0
Age_Group                  0
Purchase_Frequency_Num     0
dtype: int64

In [18]:
df.duplicated()


0       False
1       False
2       False
3       False
4       False
        ...  
3895    False
3896    False
3897    False
3898    False
3899    False
Length: 3900, dtype: bool

In [19]:
df.duplicated().sum()

np.int64(0)

In [20]:
df["Gender"].unique()

<StringArray>
['Male', 'Female']
Length: 2, dtype: str

In [21]:
df["Location"].nunique()


50

In [22]:
df["Location"].value_counts()

Location
Montana           96
California        95
Idaho             93
Illinois          92
Alabama           89
Minnesota         88
New York          87
Nevada            87
Nebraska          87
Delaware          86
Maryland          86
Vermont           85
Louisiana         84
North Dakota      83
West Virginia     81
Missouri          81
New Mexico        81
Mississippi       80
Kentucky          79
Arkansas          79
Indiana           79
Georgia           79
North Carolina    78
Connecticut       78
Maine             77
Texas             77
Tennessee         77
Ohio              77
Virginia          77
South Carolina    76
Oklahoma          75
Colorado          75
Wisconsin         75
Oregon            74
Pennsylvania      74
Washington        73
Michigan          73
Massachusetts     72
Alaska            72
Wyoming           71
New Hampshire     71
Utah              71
South Dakota      70
Iowa              69
Florida           68
New Jersey        67
Hawaii            65
Ariz

In [23]:
#on determine les noms des colonnes de type catégoriel
#df.select_dtypes(include=["str", "category"]) cette partie donne
# un data frame qu'avec les types mentionné dans include 
colonnes_cat = df.select_dtypes(include=["str", "category"]).columns
for col in colonnes_cat:
    print("\n---", col, "---")
    print("Nombre de valeurs différentes :", df[col].nunique())
    print(df[col].value_counts())
#On remarque qu'il n'y a pas d'incohérence 


--- Gender ---
Nombre de valeurs différentes : 2
Gender
Male      2652
Female    1248
Name: count, dtype: int64

--- Item Purchased ---
Nombre de valeurs différentes : 25
Item Purchased
Blouse        171
Pants         171
Jewelry       171
Shirt         169
Dress         166
Sweater       164
Jacket        163
Coat          161
Sunglasses    161
Belt          161
Sandals       160
Socks         159
Skirt         158
Shorts        157
Scarf         157
Hat           154
Handbag       153
Hoodie        151
Shoes         150
T-shirt       147
Sneakers      145
Boots         144
Backpack      143
Gloves        140
Jeans         124
Name: count, dtype: int64

--- Category ---
Nombre de valeurs différentes : 4
Category
Clothing       1737
Accessories    1240
Footwear        599
Outerwear       324
Name: count, dtype: int64

--- Location ---
Nombre de valeurs différentes : 50
Location
Montana           96
California        95
Idaho             93
Illinois          92
Alabama           89
Min

In [25]:
# ON veut vérifier si les deux colonnes sont identiques 
(df["Discount Applied"] == df["Promo Code Used"]).sum()
#Donc les colonnes sont identiques 

np.int64(3900)

In [26]:
(df["Age"]).min()

np.int64(18)

In [27]:
(df["Age"]).max()

np.int64(70)

In [28]:
df.describe()

,Customer ID,Age,Purchase Amount (USD),Review Rating,Previous Purchases,Purchase_Frequency_Num
count,3900.000000,3900.000000,3900.000000,3863.000000,3900.000000,3900.000000
mean,1950.500000,44.068462,59.764359,3.750065,25.351538,17.471282
std,1125.977353,15.207589,23.685392,0.716983,14.447125,16.809646
min,1.000000,18.000000,20.000000,2.500000,1.000000,1.000000
25%,975.750000,31.000000,39.000000,3.100000,13.000000,4.000000
50%,1950.500000,44.000000,60.000000,3.800000,25.000000,12.000000
75%,2925.250000,57.000000,81.000000,4.400000,38.000000,26.000000
max,3900.000000,70.000000,100.000000,5.000000,50.000000,52.000000


In [33]:
#Calcul de l'IQR pour les valeurs aberrantes
Q1 = df["Purchase Amount (USD)"].quantile(0.25)
Q3 = df["Purchase Amount (USD)"].quantile(0.75)

In [37]:
IQR = Q3 - Q1

In [35]:
borne_basse = Q1 - 1.5 * IQR
borne_haute = Q3 + 1.5 * IQR

In [36]:
outliers = df[
    (df["Purchase Amount (USD)"] < borne_basse) |
    (df["Purchase Amount (USD)"] > borne_haute)
]
# Cette data frame contient les valeurs aberrantes 
outliers.shape[0]


0

In [42]:
#Meme traitement d'outliers pour les autres facteurs numériques 
colonnes_outliers = [
    "Age",
    "Purchase Amount (USD)",
    "Review Rating",
    "Previous Purchases"
]
for col in colonnes_outliers:

    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)

    IQR = Q3 - Q1

    borne_basse = Q1 - 1.5 * IQR
    borne_haute = Q3 + 1.5 * IQR

    outliers = df [(df[col] < borne_basse) |( df[col] > borne_haute)]

    print(col)
    print("Borne basse :", borne_basse)
    print("Borne haute :", borne_haute)
    print("Nombre d'outliers :",outliers.shape[0])
    print()

Age
Borne basse : -8.0
Borne haute : 96.0
Nombre d'outliers : 0

Purchase Amount (USD)
Borne basse : -24.0
Borne haute : 144.0
Nombre d'outliers : 0

Review Rating
Borne basse : 1.1499999999999997
Borne haute : 6.3500000000000005
Nombre d'outliers : 0

Previous Purchases
Borne basse : -24.5
Borne haute : 75.5
Nombre d'outliers : 0



In [45]:
# On garde les lignes qui correspondent à TRUE
missing_rating = df[df["Review Rating"].isnull()]
#On va essayer de comprendre ces données 
missing_rating.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases,Age_Group,Purchase_Frequency_Num
38,39,29,Male,Dress,Clothing,37,Florida,M,Red,Winter,NaN,Yes,2-Day Shipping,Yes,Yes,44,Venmo,Every 3 Months,26-35,4
50,51,49,Male,Blouse,Clothing,28,Maryland,M,Red,Spring,NaN,Yes,Store Pickup,Yes,Yes,39,Debit Card,Bi-Weekly,46-55,26
80,81,19,Male,Sandals,Footwear,72,New York,XL,Blue,Summer,NaN,Yes,Store Pickup,Yes,Yes,24,Credit Card,Every 3 Months,18-25,4
96,97,32,Male,Gloves,Accessories,73,Delaware,S,Gold,Spring,NaN,Yes,Express,Yes,Yes,43,Venmo,Every 3 Months,26-35,4
262,263,37,Male,Pants,Clothing,40,Hawaii,M,Teal,Summer,NaN,Yes,Next Day Air,Yes,Yes,6,Debit Card,Annually,36-45,1


In [46]:
missing_rating["Gender"].value_counts()

Gender
Male    37
Name: count, dtype: int64

In [47]:
missing_rating["Subscription Status"].value_counts()


Subscription Status
Yes    20
No     17
Name: count, dtype: int64

In [48]:
missing_rating["Discount Applied"].value_counts()

Discount Applied
Yes    37
Name: count, dtype: int64

In [49]:
# 1. Supprimer la colonne redondante
df = df.drop(columns=["Promo Code Used"])

In [50]:
# 8. Sauvegarder le dataset nettoyé
df.to_csv("customer_shopping_cleaned.csv", index=False)

# 2. Analyse des relations (Descriptive)

## 2.1 Age_Group

In [51]:
df.groupby("Age_Group")["Purchase Amount (USD)"].mean()

Age_Group
18-25    60.647986
26-35    59.760108
36-45    59.305898
46-55    60.583001
56+      59.055204
Name: Purchase Amount (USD), dtype: float64

In [52]:
df.groupby("Age_Group")["Purchase_Frequency_Num"].mean()

Age_Group
18-25    16.749562
26-35    17.863881
36-45    17.674897
46-55    18.067729
56+      17.039819
Name: Purchase_Frequency_Num, dtype: float64

#### Descriptivement, les différences entre les tranches d'âge semblent faibles.

## 2.3 Gender

In [56]:
df.groupby("Gender")["Purchase Amount (USD)"].mean()

Gender
Female    60.249199
Male      59.536199
Name: Purchase Amount (USD), dtype: float64

In [57]:
df.groupby("Gender")["Purchase_Frequency_Num"].mean()

Gender
Female    17.544071
Male      17.437029
Name: Purchase_Frequency_Num, dtype: float64

#### Faible différence descriptive apparente.

## 2.4 Location

In [58]:
df.groupby("Location")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Location
Alaska            67.597222
Pennsylvania      66.567568
Arizona           66.553846
West Virginia     63.876543
Nevada            63.379310
Washington        63.328767
North Dakota      62.891566
Virginia          62.883117
Utah              62.577465
Michigan          62.095890
Tennessee         61.974026
New Mexico        61.901235
Rhode Island      61.444444
Texas             61.194805
Arkansas          61.113924
Illinois          61.054348
Mississippi       61.037500
Massachusetts     60.888889
Iowa              60.884058
North Carolina    60.794872
Wyoming           60.690141
South Dakota      60.514286
New York          60.425287
Ohio              60.376623
Montana           60.250000
Idaho             60.075269
Nebraska          59.448276
New Hampshire     59.422535
Alabama           59.112360
California        59.000000
Indiana           58.924051
Georgia           58.797468
South Carolina    58.407895
Oklahoma          58.346667
Missouri          57.913580
Hawaii     

#### Location → PA : différences descriptives visibles selon les États, à étudier statistiquement plus tard.

In [59]:
df.groupby("Location")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Location
Hawaii            22.215385
Louisiana         21.000000
Georgia           20.620253
Tennessee         19.883117
Oklahoma          19.880000
Illinois          19.847826
Indiana           19.696203
Massachusetts     19.680556
North Carolina    19.679487
Oregon            19.351351
Alaska            19.291667
New York          19.275862
Kentucky          18.632911
Virginia          18.571429
Kansas            18.523810
Wyoming           18.507042
Rhode Island      18.428571
Minnesota         18.386364
Maryland          18.383721
Pennsylvania      18.297297
Alabama           18.213483
Iowa              17.869565
New Mexico        17.604938
Mississippi       17.550000
Ohio              17.454545
Nevada            17.333333
Texas             17.298701
West Virginia     17.259259
Arkansas          16.974684
California        16.726316
Nebraska          16.551724
South Dakota      16.442857
Arizona           16.307692
Maine             16.285714
Montana           16.260417
Florida    

#### Pour Location → FA, les écarts sont également plus visibles

## 2.5 Subscription status

In [60]:
df.groupby("Subscription Status")["Purchase Amount (USD)"].mean()

Subscription Status
No     59.865121
Yes    59.491928
Name: Purchase Amount (USD), dtype: float64

In [61]:
df.groupby("Subscription Status")["Purchase_Frequency_Num"].mean()

Subscription Status
No     17.292589
Yes    17.954416
Name: Purchase_Frequency_Num, dtype: float64

#### Faible différence descriptive apparente.

## 2.6 Previous purchases 

In [62]:
df["Previous Purchases"].corr(df["Purchase Amount (USD)"])

np.float64(0.008063412270587633)

#### Previous Purchases → PA : pratiquement aucune relation linéaire apparente.

In [63]:
df["Previous Purchases"].corr(df["Purchase_Frequency_Num"])

np.float64(0.004399595212724948)

#### Previous Purchases → FA : pratiquement aucune relation linéaire apparente.

## 2.7 Item purchased 

In [64]:
df.groupby("Item Purchased")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Item Purchased
T-shirt       62.911565
Boots         62.625000
Dress         62.168675
Shoes         61.600000
Shirt         61.136095
Scarf         60.898089
Blouse        60.877193
Hat           60.876623
Jeans         60.870968
Gloves        60.550000
Backpack      60.391608
Shorts        60.082803
Sunglasses    59.931677
Belt          59.844720
Sneakers      59.551724
Skirt         59.506329
Pants         59.005848
Jewelry       58.538012
Socks         58.188679
Hoodie        58.059603
Handbag       57.888889
Sweater       57.695122
Coat          57.608696
Sandals       57.500000
Jacket        56.742331
Name: Purchase Amount (USD), dtype: float64

#### Pour Item Purchased → PA, on voit des écarts descriptifs plus visibles

In [65]:
df.groupby("Item Purchased")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Item Purchased
Sweater       20.646341
Shorts        19.484076
Boots         19.131944
Skirt         19.075949
Coat          18.739130
Backpack      18.489510
Sunglasses    18.118012
Gloves        18.100000
Sneakers      18.096552
Hoodie        18.086093
Jewelry       18.035088
Shirt         17.627219
Sandals       17.450000
Pants         17.444444
Belt          17.130435
Shoes         17.093333
Scarf         16.808917
Dress         16.692771
T-shirt       16.680272
Jacket        16.613497
Hat           16.162338
Handbag       15.549020
Blouse        15.473684
Jeans         15.177419
Socks         14.660377
Name: Purchase_Frequency_Num, dtype: float64

#### Pour Item Purchased → FA, on voit des écarts descriptifs plus visibles?

## 2.8 Category

In [66]:
df.groupby("Category")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Category
Footwear       60.255426
Clothing       60.025331
Accessories    59.838710
Outerwear      57.172840
Name: Purchase Amount (USD), dtype: float64

#### Category → PA : les montants moyens sont assez proches, avec Outerwear légèrement plus bas. L'écart descriptif reste modéré.

In [67]:
df.groupby("Category")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)


Category
Footwear       17.921536
Outerwear      17.669753
Clothing       17.405872
Accessories    17.293548
Name: Purchase_Frequency_Num, dtype: float64

#### Category → FA : différences descriptives faibles. Les clients associés aux différentes catégories ont une fréquence générale d'achat assez similaire.

In [68]:
df["Category"].value_counts()

Category
Clothing       1737
Accessories    1240
Footwear        599
Outerwear       324
Name: count, dtype: int64

In [69]:
df.groupby("Category")["Purchase Amount (USD)"].sum().sort_values(ascending=False)

Category
Clothing       104264
Accessories     74200
Footwear        36093
Outerwear       18524
Name: Purchase Amount (USD), dtype: int64

#### Clothing est la catégorie la plus importante commercialement dans le dataset en volume et en chiffre d'affaires, malgré un montant moyen d'achat similaire aux autres grandes catégories.

## 2.9 Size 

In [71]:
df.groupby("Size")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Size
S     61.037707
XL    60.090909
M     59.924217
L     58.563153
Name: Purchase Amount (USD), dtype: float64

#### Size → PA : différences descriptives plutôt faibles.

In [72]:
df.groupby("Size")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Size
M     17.682051
XL    17.603730
S     17.580694
L     16.997151
Name: Purchase_Frequency_Num, dtype: float64

#### Size → FA : différences descriptives faibles.

## 2.10 Color

In [73]:
df.groupby("Color")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Color
Green        65.704142
White        62.640845
Gray         62.490566
Cyan         61.891566
Violet       61.716867
Gold         61.007246
Orange       60.889610
Teal         60.808140
Charcoal     60.633987
Pink         60.588235
Beige        60.414966
Purple       60.013245
Maroon       59.525316
Red          59.317568
Yellow       59.241379
Peach        59.187919
Lavender     59.129252
Brown        59.063830
Black        58.401198
Olive        58.146893
Magenta      57.131579
Blue         56.953947
Silver       56.832370
Indigo       56.251701
Turquoise    55.613793
Name: Purchase Amount (USD), dtype: float64

#### Color → PA : écarts descriptifs visibles selon les couleurs. Cette variable pourra être approfondie statistiquement plus tard.

In [74]:
df.groupby("Color")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Color
Cyan         19.289157
Turquoise    19.068966
Gray         18.786164
Violet       18.692771
Lavender     18.687075
Indigo       18.517007
Purple       18.470199
Yellow       18.316092
Olive        18.067797
Silver       17.826590
Red          17.817568
Charcoal     17.699346
Orange       17.519481
Brown        17.390071
Black        17.365269
Gold         17.130435
Blue         16.967105
Peach        16.812081
Beige        16.727891
White        16.429577
Pink         16.418301
Green        16.171598
Teal         15.720930
Magenta      15.407895
Maroon       15.367089
Name: Purchase_Frequency_Num, dtype: float64

#### Color → FA : on observe des écarts descriptifs visibles, mais moins marqués que pour certaines variables comme Location.

## 2.11 Season 

In [76]:
df.groupby("Season")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Season
Fall      61.556923
Winter    60.357364
Spring    58.737738
Summer    58.405236
Name: Purchase Amount (USD), dtype: float64

#### Season → PA : on observe un écart descriptif modéré, avec un PA moyen plus élevé en automne et plus faible en été.

In [77]:
df.groupby("Season")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Season
Winter    18.217302
Spring    17.906907
Fall      16.876923
Summer    16.863874
Name: Purchase_Frequency_Num, dtype: float64

#### Season → FA : différences descriptives plutôt faibles à modérées, avec une fréquence moyenne plus élevée en hiver.

## 2.12 Dicount applied 

In [78]:
df.groupby("Discount Applied")["Purchase Amount (USD)"].mean()

Discount Applied
No     60.130454
Yes    59.279070
Name: Purchase Amount (USD), dtype: float64

#### Discount Applied → PA : différence descriptive faible. Les achats avec réduction ont un montant moyen légèrement inférieur, mais l'écart est petit.

In [79]:
df.groupby("Discount Applied")["Purchase_Frequency_Num"].mean()

Discount Applied
No     17.408007
Yes    17.555158
Name: Purchase_Frequency_Num, dtype: float64

#### FA → très faible différence

## 2.13 Shipping type

In [80]:
df.groupby("Shipping Type")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Shipping Type
2-Day Shipping    60.733652
Express           60.475232
Free Shipping     60.410370
Store Pickup      59.893846
Next Day Air      58.631173
Standard          58.460245
Name: Purchase Amount (USD), dtype: float64

#### Shipping Type → PA : différences descriptives plutôt faibles.

In [81]:
df.groupby("Shipping Type")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Shipping Type
Express           17.981424
Free Shipping     17.638519
Standard          17.600917
Next Day Air      17.387346
2-Day Shipping    17.263158
Store Pickup      16.944615
Name: Purchase_Frequency_Num, dtype: float64

#### Shipping Type → FA : différences descriptives faibles.

## 2.14 Payment method

In [82]:
df.groupby("Payment Method")["Purchase Amount (USD)"].mean().sort_values(ascending=False)

Payment Method
Debit Card       60.915094
Credit Card      60.074516
Bank Transfer    59.712418
Cash             59.704478
PayPal           59.245199
Venmo            58.949527
Name: Purchase Amount (USD), dtype: float64

#### Payment Method → PA : différences descriptives faibles.

In [83]:
df.groupby("Payment Method")["Purchase_Frequency_Num"].mean().sort_values(ascending=False)

Payment Method
Cash             18.310448
Credit Card      17.940387
Bank Transfer    17.374183
PayPal           17.203840
Debit Card       16.982704
Venmo            16.957413
Name: Purchase_Frequency_Num, dtype: float64

#### Payment Method → FA : différences descriptives faibles à modérées, avec une fréquence moyenne un peu plus élevée chez les clients payant en Cash.

## 3. Analyse des relations(Statistique)

## 3.1 Les indicateurs du comportement 

#### L'ANOVA pose essentiellement :

#### H₀ : les moyennes de PA des 5 groupes d'âge sont égales.
#### H₁ : au moins une moyenne est différente.
#### p-value < 0.05 → on rejette H₀

In [85]:
from scipy.stats import f_oneway

In [87]:
groupes = df.groupby("Age_Group", observed=True)["Purchase Amount (USD)"]


In [88]:
groupes

In [89]:
groupes_age = [groupe for _, groupe in groupes]
groupes_age
#une liste de Series Pandas ( inedex et valeur)

[1       64
 3       90
 16      36
 20      51
 24      22
         ..
 3830    27
 3848    60
 3865    42
 3868    29
 3893    64
 Name: Purchase Amount (USD), Length: 571, dtype: int64,
 7       34
 8       97
 11      68
 21      62
 23      88
         ..
 3879    63
 3882    95
 3883    38
 3890    81
 3892    86
 Name: Purchase Amount (USD), Length: 742, dtype: int64,
 4       49
 26      20
 32      67
 34      91
 44      53
         ..
 3887    34
 3888    69
 3891    30
 3895    28
 3898    77
 Name: Purchase Amount (USD), Length: 729, dtype: int64,
 0       53
 2       73
 5       20
 10      34
 17      38
         ..
 3884    29
 3885    64
 3896    49
 3897    33
 3899    81
 Name: Purchase Amount (USD), Length: 753, dtype: int64,
 6       85
 9       31
 12      72
 13      51
 14      53
         ..
 3876    85
 3878    42
 3881    25
 3889    65
 3894    78
 Name: Purchase Amount (USD), Length: 1105, dtype: int64]

In [90]:
resultat = f_oneway(*groupes_age)

print(resultat)

F_onewayResult(statistic=np.float64(0.7392979305450704), pvalue=np.float64(0.565075850776863))


#### On n’a pas détecté de différence significative du PA moyen selon les groupes d’âge.

In [91]:
from scipy.stats import f_oneway, ttest_ind

facteurs = [
    "Age_Group",
    "Gender",
    "Location",
    "Subscription Status",
    "Item Purchased",
    "Category",
    "Size",
    "Color",
    "Season",
    "Shipping Type",
    "Discount Applied",
    "Payment Method"
]

cibles = ["Purchase Amount (USD)", "Purchase_Frequency_Num"]

for facteur in facteurs:
    for cible in cibles:

        groupes = [
            groupe[cible].dropna()
            for _, groupe in df.groupby(facteur, observed=True)
        ]

        if len(groupes) == 2:
            stat, p = ttest_ind(
                groupes[0],
                groupes[1],
                equal_var=False
            )
            test = "Test t"

        else:
            stat, p = f_oneway(*groupes)
            test = "ANOVA"

        print(
            facteur,
            "→",
            cible,
            "|",
            test,
            "| p-value =",
            round(p, 4)
        )

Age_Group → Purchase Amount (USD) | ANOVA | p-value = 0.5651
Age_Group → Purchase_Frequency_Num | ANOVA | p-value = 0.5186
Gender → Purchase Amount (USD) | Test t | p-value = 0.3778
Gender → Purchase_Frequency_Num | Test t | p-value = 0.8527
Location → Purchase Amount (USD) | ANOVA | p-value = 0.0973
Location → Purchase_Frequency_Num | ANOVA | p-value = 0.4921
Subscription Status → Purchase Amount (USD) | Test t | p-value = 0.6603
Subscription Status → Purchase_Frequency_Num | Test t | p-value = 0.2813
Item Purchased → Purchase Amount (USD) | ANOVA | p-value = 0.7354
Item Purchased → Purchase_Frequency_Num | ANOVA | p-value = 0.303
Category → Purchase Amount (USD) | ANOVA | p-value = 0.2252
Category → Purchase_Frequency_Num | ANOVA | p-value = 0.8873
Size → Purchase Amount (USD) | ANOVA | p-value = 0.1882
Size → Purchase_Frequency_Num | ANOVA | p-value = 0.7607
Color → Purchase Amount (USD) | ANOVA | p-value = 0.0666
Color → Purchase_Frequency_Num | ANOVA | p-value = 0.836
Season → Pur

#### Seule la saison est significativement associée au PA ; aucun facteur testé n’est significativement associé à la FA.

## 3.2 Les indicateurs Business 

##### c'est cette partie qui nous aide à les determiner:"comment exploiter les résultats pour améliorer les ventes et la fidélisation"

In [92]:
business_category = df.groupby("Category").agg(
    Occurrences=("Customer ID", "count"),
    CA=("Purchase Amount (USD)", "sum"),
    PA_Moyen=("Purchase Amount (USD)", "mean")
)

business_category

,Occurrences,CA,PA_Moyen
Category,,,
Accessories,1240,74200,59.838710
Clothing,1737,104264,60.025331
Footwear,599,36093,60.255426
Outerwear,324,18524,57.172840


In [93]:
df["Subscription Status"].value_counts(normalize=True) * 100

Subscription Status
No     73.0
Yes    27.0
Name: proportion, dtype: float64

In [94]:
df["Purchase_Frequency_Num"].mean()

np.float64(17.471282051282053)

In [95]:
df["Purchase Amount (USD)"].sum()

np.int64(233081)

In [96]:
df["Purchase Amount (USD)"].mean()

np.float64(59.76435897435898)

#### À ce stade, on a les KPI business principaux :
#### CA total : 233 081 
#### Nombre d'occurrences : 3 900
#### PA moyen : ~59,76 
#### Taux d'abonnement : 27 %
#### Fréquence moyenne : ~17,47 achats/an